hudi-examples/src/main/java/org/apache/hudi/examples/java/HoodieJavaWriteClientExample.java

```diff
@@ -93,7 +94,8 @@ public static void main(String[] args) throws Exception {
     List<HoodieRecord<HoodieAvroPayload>> recordsSoFar = new ArrayList<>(records);
     List<HoodieRecord<HoodieAvroPayload>> writeRecords =
         recordsSoFar.stream().map(r -> new HoodieRecord<HoodieAvroPayload>(r)).collect(Collectors.toList());
-    client.upsert(writeRecords, newCommitTime);
+    List<WriteStatus> insertStatus = client.upsert(writeRecords, newCommitTime);
+    client.commit(newCommitTime,insertStatus);
```

Comment:
would we just change the BaseJavaCommitActionExecutor#execute method to add `updateIndexAndCommitIfNeeded` method and align with BaseSparkCommitActionExecutor?

Reply:
@leesf many thanks for your review. You suggestion make more sense and i have change the code accordingly.